### ISSUE TITLE
Enabling the layer normalization decrease the performance of the model

### ISSUE THREAD

**Issue by AzizCode92** [vrenkens/nabu#44]
I have tried to add layer normalization to the layers.py by setting layer_norm to True as a default value.
I used the "listener.py" as my encoder. 
Those modifications decreased the performance of the model ( training time is longer, both accuracy and WER decreased dramatically).
I have seen other users complaining about such problems with `tf.contrib.rnn.LayerNormBasicLSTMCell`   
in [stackoverflow](https://stackoverflow.com/questions/45150101/why-is-layernormbasiclstmcell-much-slower-and-less-accurate-than-lstmcell)
Am I missing something else?

**Comment by vrenkens** [comment]
I have experienced the same issues... Maybe it is better to use the `tf.contrib.rnn.LSTMCell` instead

**Comment by AzizCode92** [comment]
Thank you for your answer. That's a strange behaviour.
I'm thinking of using layer normalization in the level of the encoder/decoder outpout layer instead on focusing on using the tf.contrib.rnn.NormLayerBasicLSTMCell.

By the way I have opened an issue on tensorflow [issue](https://github.com/tensorflow/tensorflow/issues/21929) for this problem.

**Comment by AzizCode92** [comment]
I was the last days looking for the reasons why we got worse performance when we apply layerNormalization and I come through one [paper](https://arxiv.org/pdf/1801.05134.pdf) where the authors explain why **BatchNormalization** and **dropout** give poor performance when applied together. Although they talked about BatchNormalization I think it will apply also for Layer Normalization.
It might be an explanation for such behaviour

**Comment by AzizCode92** [comment]
#Update : I have considered this [pull request](https://github.com/MycChiu/fast-LayerNorm-TF) and created a custom LSTMCell with layer_normalization and dropout. 
The validation loss decreased in few steps from  to 

> WORKER 0: validating model
WORKER 0: validation loss: 2.573561
WORKER 0: validating model
WORKER 0: validation loss: 0.965548
WORKER 0: validating model
WORKER 0: validation loss: 0.990690
WORKER 0: validation loss is worse
WORKER 0: validating model
WORKER 0: validation loss: 0.990694
WORKER 0: validation loss is worse
but then stays there and didn't improve further.
I used the listener.py and modified where I deleted the dropout (since I included in the LSTMCells level)
Any Idea how to improve this?
I have added also layer_normalization between the layers in the listener but I got the same results.

**Comment by AzizCode92** [comment]
Hi Vincent, I have used the TIMIT dataset and I used the recipe of LAS and it worked fine for my case. 
I have attached my validation loss (the blue one is without layer normalization and the orange one is with layer normalization).
![screen shot 2018-12-03 at 12 56 53](https://user-images.githubusercontent.com/19540527/49372461-0cfe0b80-f6fb-11e8-9e8e-3d870e108caf.png)

Why it didn't work for you case?
P.S: I used the KALDI-ASR project to process the dataset

with best regards,
Aziz

**Comment by AzizCode92** [comment]
Previously, I have pointed that it didn't work with the librispeech but I do have doubts that it is due to some mistake I might have done during the processing.
